pkg/controller/services.go

```diff
@@ -535,6 +599,105 @@ func (cont *AciController) updateServiceDeviceInstance(key string,
 	return nil
 }
 
+func (cont *AciController) updateServiceDeviceInstanceSnat(key string) error {
```

Comment:
There are at least parts of the function updateServiceDeviceInstance which are very similar to this one. We should attempt to refactor and avoid code duplication. This can be done as a follow up patch.

Reply:
Noted!